docs/en/integrations/build-integration.asciidoc

```diff
@@ -208,20 +208,88 @@ apache
 │   │   └───elasticsearch/ingest_pipeline
 │   │   │      default.yml
 │   │   └───fields
-│   │          agent.yml <1>
-│   │          base-fields.yml <2>
-│   │          ecs.yml <3>
-│   │          fields.yml <4>
+│   │          agent.yml
+│   │          base-fields.yml
+│   │          ecs.yml
+│   │          fields.yml
 │   └───error
 │   └───status
 ----
-<1> `agent.yml` fields the Elastic agent uses
-<2> `base-fields.yml` never changes and is required for all integrations
-<3> Defines the relevant ECS fields
-<4> Custom Apache access log fields
 
+=== agent.yml
+The `agent.yml` file defines fields used by default processors, this file usually remains unchanged between integrations.
+
+=== base-fields.yml
+The fields defined in this file are placeholders for the datastream subfields which, in turn, receive their value from the Elastic agent.
+In addition, the fields `event.module` and `event.dataset` are also present; the values in these fields depend on the integration.
+For example, in the Apache integration (more specifically, for the access dataset) that would be:
+* event.module: apache
+* event.dataset: apache.access
+
+=== ecs.yml
+This file specifies every Elastic Common Schema (ECS) field used by the integration that is not defined in the files `agent.yml` or `base-fields.yml` files.
+To avoid redefining every ECS field in every integration, we can specify an existing field as external. For example:
+
+[source,yaml]
+----
+- external: ecs
+  name: client.ip
+- external: ecs
+  name: destination.domain
+----
+
+The `external: ecs` definition instructs the `elastic-package` command line tool to refer to an external ECS reference to resolve specific fields. By default it looks at the https://raw.githubusercontent.com/elastic/ecs/v8.6.0/generated/ecs/ecs_nested.yml[ECS reference] file hosted on Github. This external reference file is determined by a Git reference found in the `_dev/build/build.yml` file, located starting from the root of the package directory.
+The `build.yml` file should look like the following:
++
+[source,yaml]
+----
+dependencies:
+  ecs:
+    reference: git@v8.6.0
+----
+
+In addition to this approach, we support three more methods to import the ECS, described below.
```

Comment:
@jsoriano , would it be good to suggest a preferred method here?

Reply:
We could provide kind of a tree of decision:
* If the package only supports versions starting on 8.13.0, it doesn't need to import or define any ECS field, they are going to be provided by the stack.
* If the package supports older versions, and uses ECS mappings, it needs to import them, using `import_mappings: true` would be enough.
* If the package needs to overwrite some ECS mapping, these fields can  be imported with `external`. For the rest of fields the package would also use any of the above methods depending on the versions of the stack supported.

When does a package need to overwrite ECS mappings with `external: ecs` fields? Some examples:
* In cases where the provided dynamic mappings are not specific enough and there are mapping issues (not known example now, but could happen in theory).
* In cases where we want to provide values for `constant_keyword`s.
* In other cases where we would want to do fine-grained optimizations, like selecting between different text or numeric types.
* ...

Reply:
This information is super useful for an integration developer.

Reply:
One more question about this for now; If developers go the `ecs@mapping` route, do they still need to define fields that have historically been defined in agent.yml?
My gut feeling says no, but I want confirmation before I'm going to put it down in writing.

Reply:
> One more question about this for now; If developers go the `ecs@mapping` route, do they still need to define fields that have historically been defined in agent.yml?

It will depend on whether the field is part of ECS or not. Many fields will be, but some of them are not. For example the apache package defines `input.type` in the `agent.yml` file, but this field is not part of ECS.

Reply:
> Another question popped up, do we need to put the following in the `manifest.yml` file of the dataset in the integration to make dynamic mappings work?

It should not be needed to explicitly enable dynamic mappings, I think that default behavior is fine for the mappings provided for ECS.